firedrake/slate/slate.py

```diff
@@ -567,7 +568,7 @@ class Action(TensorOp):
     :arg function: a :class:`firedrake.Function` object.
     """
 
-    prec = 2
+    prec = 3
```

Comment:
I see you increment most of these precedence values. It scales better if you use the Martin/UFL approach (see `ufl.precedence`).